File: BOJ/28702.rs
use std::io::*;

fn main() {
    let input = read_to_string(stdin()).unwrap();
    let lines = input.lines();

    let (i, line) = lines
        .enumerate()
        .find(|&(_, line)| line.parse::<usize>().is_ok())
        .unwrap();

    let fizz_buzz = |x: usize| {
        if x % 15 == 0 {
            String::from("FizzBuzz")
        } else if x % 3 == 0 {
            String::from("Fizz")
        } else if x % 5 == 0 {
            String::from("Buzz")
        } else {
            x.to_string()
        }
    };

    let output = fizz_buzz(line.parse::<usize>().unwrap() + 3 - i);

    print!("{output}");
}

File: BOJ/28702.cpp
#include <iostream>

using namespace std;

string FizzBuzz(int num) {
  if (num % 15 == 0) return "FizzBuzz";
  if (num % 3 == 0) return "Fizz";
  if (num % 5 == 0) return "Buzz";
  return to_string(num);
}

int main() {
  ios_base::sync_with_stdio(false);
  cin.tie(nullptr);

  for (int i = 3; i > 0; i--) {
    string s;
    cin >> s;
    if (isdigit(s[0])) {
      cout << FizzBuzz(stoi(s) + i);
      return 0;
    }
  }

  return 0;
}
